python/ray/serve/batching.py

```diff
@@ -228,13 +253,18 @@ async def _process_batches(self, func: Callable) -> None:
                     results = await func_future
                     self._validate_results(results, len(batch))
                     for result, future in zip(results, futures):
-                        # If the client has disconnected, the future is canceled.
-                        # We should only set result when the connection is still live.
-                        if not future.cancelled():
-                            future.set_result(result)
+                        _set_result_if_not_done(future, result)
                 except Exception as e:
                     for future in futures:
-                        future.set_exception(e)
+                        _set_exception_if_not_done(future, e)
+        except Exception as e:
+            for future in futures:
+                _set_exception_if_not_done(future, e)
+
+            logger.exception(
```

Comment:
Let's log first, then set futures (in case there are any failures to make sure we at least will get the log)

Reply:
Makes sense, I logged first.

Reply:
Wait, we now have exception handling on 3 levels

 - _process_batches
 - _process_batch
 - _consume_func_generator and _assign_func_results

I think we can drop one in _process_batch

Reply:
We still want it in `_process_batch` due to [this point](https://github.com/ray-project/ray/pull/42631#discussion_r1465256415). We could probably remove it from `_process_batches`, but I'd prefer to have one `try-except` block that covers everything.